client/src/components/History/GanttVisualisation/Gantt.vue

```diff
@@ -0,0 +1,639 @@
+import BootstrapVue from "bootstrap-vue";
+import moment from "moment";
+import Vue from "vue";
+
+Vue.use(BootstrapVue);
+
+export default {
+    name: "Gantt",
+    components: {
+        LoadingSpan,
+        EmptyHistory,
+        CurrentUser,
+        UserHistories,
+        DatePicker,
+        BModal
+    },
+    setup() {
+        // Store
+        const piniaStore = useHistoryItemsStore();
+
+        // Refs
+        const historyId = ref(store.getters["history/currentHistoryId"]);
+        const accountingArray = ref([]);
+        const accountingArrayMinutes = ref([]);
+        const historyItems = ref(piniaStore.items[historyId.value]);
+        const gantt = ref();
+        const ganttView = ref("Day");
+        const noMinuteJobs = ref(false);
+        const currentlyProcessing = ref(false);
+        const isLoading = ref(true);
+        const emptyHistory = ref(false);
+        const dateTimeVal = ref(new Date().toLocaleString("en-GB"));
+        const start_time = ref(null);
+        const end_time = ref(null);
+        const empty_metrics = ref(0);
+        const noMetrics = ref(false);
+        const message = ref("");
+        const date = ref(moment().format("YYYY-MM-DD HH:mm:ss"));
+
+        // Hooks
+        onMounted(() => {
+            createKeyedColorForButtons();
+            getData();
+        });
+
+        // Watchers
+        watch(
+            () => store.getters["history/currentHistoryId"],
+            (newHistoryId, oldHistoryId) => {
+                if (newHistoryId !== oldHistoryId) {
+                    // Making currently processing false so that when you switch to a new History, we can re-fetch the historyContents to refresh the GANTT
+                    currentlyProcessing.value = false;
+                    historyId.value = newHistoryId;
+                    empty_metrics.value = 0;
+                    noMetrics.value = false;
+                    message.value = "";
+                    clearGantt();
+                    clearPopups();
+                    ganttView.value = "Day";
+                    createKeyedColorForButtons();
+                }
+            }
+        );
+
+        watch(historyContent, async (newContent, oldContent) => {
+            newContent.then((res) => {
+                if (res && res.length > 0 && !currentlyProcessing.value) {
+                    emptyHistory.value = false;
+                    historyItems.value = res;
+                    getData();
+                } else if (res && res.length == 0 && !currentlyProcessing.value) {
+                    isLoading.value = false;
+                    emptyHistory.value = true;
+                }
+            });
+            createKeyedColorForButtons();
+        });
+
+        watch(emptyHistory, (newEmpty, oldEmpty) => {
+            if (newEmpty === true) {
+                clearGantt();
+            }
+        });
+
+        watch(ganttView, (newval, oldval) => {
+            if (oldval == "Minute") {
```

Comment:
```suggestion
            if (oldval == "Minute") {
            clearPopups();
```
Use `clearPopups();` here instead of calling it on every change function